### PR TITLE
[tvnoe] Fixing/changing the extractor for TV Noe (Czech Christian TV)

### PR DESCRIPTION
**Pull request by jtrmal**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

The extractor was ancient. I updated it to work with the current version of the web

### CODE REVIEW

**Comment by jtrmal** [comment]
Any by ancient I mean it didn't work (not even URL matched with the current web). I also checked the original link in TEST and that didn't work either, so I didn't see any reason why to keep the original extractor.

**Comment by dstftw** [comment]
Read coding conventions.

**Comment by jtrmal** [comment]
Thanks for the input -- I tried to address it in a way... will be grateful for suggestion how else it should be done.

File: youtube_dl/extractor/tvnoe.py [hunk 1]
```diff
@@ -4,45 +4,54 @@
...
 class TVNoeIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?tvnoe\.cz/video/(?P<id>[0-9]+)'
+    _VALID_URL = r'https?://(?:www\.)?tvnoe\.cz/porad/(?P<id>[-0-9a-z]+)'
     _TEST = {
-        'url': 'http://www.tvnoe.cz/video/10362',
-        'md5': 'aee983f279aab96ec45ab6e2abb3c2ca',
+        'url': 'https://www.tvnoe.cz/porad/26011-terra-santa-news-13-11-2019',
         'info_dict': {
-            'id': '10362',
+            'id': '26011-terra-santa-news-13-11-2019',
```
Comment (dstftw) [inline youtube_dl/extractor/tvnoe.py]: Not an id.

File: youtube_dl/extractor/tvnoe.py [hunk 2]
```diff
@@ -4,45 +4,54 @@
...
-        info_dict.update({
+        formats = []
+        hls_url = self._search_regex(
+            r"\s*src:\s*\'(?P<url>https?://[^\']+playlist.m3u8)\',", webpage, 'm3u8', fatal=False)
+        if hls_url:
+            dash_url = self._search_regex(
+                r"\s*src:\s*\'(?P<url>https?://[^\']+manifest.mpd)\',", webpage, 'mpd', fatal=False)
+        else:
+            dash_url = self._search_regex(
+                r"\s*src:\s*\'(?P<url>https?://[^\']+manifest.mpd)\',", webpage, 'mpd')
```
Comment (dstftw) [inline youtube_dl/extractor/tvnoe.py]: `_parse_json`.

File: youtube_dl/extractor/tvnoe.py [hunk 3]
```diff
@@ -4,45 +4,54 @@
...
+            formats.extend(self._extract_mpd_formats(
+                dash_url, video_id, mpd_id='dash', fatal=False))
+        if hls_url:
+            if formats:
+                formats.extend(self._extract_m3u8_formats(
+                    hls_url, video_id, ext='mp4', m3u8_id='hls', fatal=False))
+            else:
+                formats.extend(self._extract_m3u8_formats(
+                    hls_url, video_id, ext='mp4', m3u8_id='hls'))
+
```
Comment (dstftw) [inline youtube_dl/extractor/tvnoe.py]: DRY.

File: youtube_dl/extractor/tvnoe.py [hunk 4]
```diff
@@ -4,45 +4,54 @@
...
+        self._sort_formats(formats)
+        title = clean_html(self._search_regex(
+            r"<h2>(?P<title>.*)<\/h2>", webpage, 'title'))
+        series = clean_html(self._search_regex(
+            r"<h1>(?P<series>.*)<\/h1>", webpage, 'series'))
```
Comment (dstftw) [inline youtube_dl/extractor/tvnoe.py]: Do not capture empty string. Single quotes.

Comment (dstftw) [inline youtube_dl/extractor/tvnoe.py]: Must not be fatal. Please bother to finally read coding conventions.

**Pull request closed without merging**